### PR TITLE
SDMODEL H7 V2

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

**Comment by okok125** [comment]
Thank You . Andy

**Comment by okok125** [comment]
[README.md.txt](https://github.com/ArduPilot/ardupilot/files/15213368/README.md.txt)

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
rather than me correcting the readme, why not just make the README.md have the two  board images that are different from the V1, mention the addition of the compass, and refer directly to the V1 readme file on github....will avoid ommisissions and mistakes

**Comment by okok125** [comment]
Change the default value for UART4 to GPS and for UART3 to Telem3. Added built-in IST8310 compass.
It is still possible to add a 2nd compass via SCL and SDA.
[README.md.txt](https://github.com/ArduPilot/ardupilot/files/15213606/README.md.txt)

**Comment by okok125** [comment]
> rather than me correcting the readme, why not just make the README.md have the two board images that are different from the V1, mention the addition of the compass, and refer directly to the V1 readme file on github....will avoid ommisissions and mistakes

Everything follows your instructions

File: libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,7 @@
+include ../KakuteH7-bdshot/hwdef.dat
+
+# board ID for firmware load
+APJ_BOARD_ID AP_HW_SDMODELH7V2
+
+# builtin compass on Holybro 6X
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/hwdef.dat:6]: Bad comment?

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/hwdef.dat]: Fixed thanks

**Comment by andyp1per** [comment]
README updated

File: libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+![SDH7V2](H7V1_0502.png)
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (Telem1) (MSP DisplayPort)
+ - SERIAL2 -> UART2 (Telem2) (connected to internal BT module)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md:27]:
```suggestion
 - SERIAL2 -> UART2 (Telem2) (connected to internal BT module, not useable by ArduPilot)
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
Andy, I can make the readme changes if you wish

File: libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/defaults.parm [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# setup for LEDs on chan9
+SERVO9_FUNCTION 120
+SERIAL7_PROTOCOL 16
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/defaults.parm:3]: Yes they do

File: libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (Telem1) (MSP DisplayPort)
+ - SERIAL2 -> UART2 (Telem2) (connected to internal BT module, not useable by ArduPilot)
+ - SERIAL3 -> UART3 (Telem3)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md:28]:
```suggestion
 - SERIAL3 -> UART3 (GPS)
```

File: libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (Telem1) (MSP DisplayPort)
+ - SERIAL2 -> UART2 (Telem2) (connected to internal BT module, not useable by ArduPilot)
+ - SERIAL3 -> UART3 (GPS)
+ - SERIAL4 -> UART4 (GPS)
+ - SERIAL5 -> not available
+ - SERIAL6 -> UART6 (RX6 in RCinput, ALT config to use as UART input)
+ - SERIAL7 -> UART7
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/SDMODELH7V2/README.md:32]:
```suggestion
 - SERIAL7 -> UART7 RX pin only, ESC telem)
```

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**